File: lib/item.dart
import 'package:json/json.dart' show JsonDecodable;

@JsonDecodable()
class Item {
  String added_at = '';
  TrackDetails track = TrackDetails();
}

@JsonDecodable()
class TrackDetails {
  String id = '';
  String name = '';
  int track_number = 0;
  ExternalUrls external_urls = ExternalUrls();
  List<Artist> artists = List.empty();
  Album album = Album();

  TrackDetails();
}

@JsonDecodable()
class ExternalUrls {
  String spotify = '';

  ExternalUrls();
}

@JsonDecodable()
class Artist {
  String name = '';
}

@JsonDecodable()
class Album {
  int total_tracks = 0;
  String name = '';
  String release_date = '';
  List<Image> images = List.empty();

  Album();
}

@JsonDecodable()
class Image {
  String url = '';
}

File: src/item.rs
use serde::Deserialize;

#[derive(Debug, Deserialize)]
pub struct Item {
    pub added_at: String,
    pub track: TrackDetails,
}

#[derive(Debug, Deserialize)]
pub struct TrackDetails {
    pub id: String,
    pub name: String,
    pub track_number: u32,
    pub external_urls: ExternalUrls,
    pub artists: Vec<Artist>,
    pub album: Album,
}

#[derive(Debug, Deserialize)]
pub struct ExternalUrls {
    pub spotify: String,
}

#[derive(Debug, Deserialize)]
pub struct Artist {
    pub name: String,
}

#[derive(Debug, Deserialize)]
pub struct Album {
    pub total_tracks: u32,
    pub name: String,
    pub release_date: String,
    pub images: Vec<Image>,
}

#[derive(Debug, Deserialize)]
pub struct Image {
    pub url: String,
}
